pika/channel.py

```diff
@@ -253,20 +246,19 @@ def basic_cancel(self, callback=None, consumer_tag='', nowait=False):
             # will persist until the channel is closed.
             del self._consumers[consumer_tag]
 
-        if callback is not None:
-            if nowait:
-                raise ValueError('Cannot pass a callback if nowait is True')
+        if callback:
             self.callbacks.add(self.channel_number, spec.Basic.CancelOk,
                                callback)
 
         self._cancelled.add(consumer_tag)
 
         self._rpc(spec.Basic.Cancel(consumer_tag=consumer_tag, nowait=nowait),
                   self._on_cancelok if not nowait else None,
-                  [(spec.Basic.CancelOk, {'consumer_tag': consumer_tag})] if
-                  nowait is False else [])
+                  [(spec.Basic.CancelOk, { 'consumer_tag': consumer_tag })] if
+                  not nowait else [])
 
-    def basic_consume(self, consumer_callback,
+    def basic_consume(self,
+                      callback,
                       queue='',
```

Comment:
`queue` shouldn't have a default value either, since you always need to pass the name of the queue to be consued, right?